API.md

```diff
@@ -437,6 +438,27 @@ append(4, [1, 2, 3]); // Iterable[1, 2, 3, 4]
 
 See [append](#append)
 
+### distinct
+
+**distinct(selector, [source](#wrappable))**  
+**distinct([source](#wrappable))**  
+**__distinct([source](#iterable), ?selector)**  
+
+Yields only the first occurrance of each value from `source`. An optional `selector` can be used to specify a transform function which returns a key to distinguish the values.
```

Comment:
What do you think is the best way to handle stuff like this? I guess I could just edit your branch if you've granted me perms. Unfortunately it seems like as a maintainer it's difficult to tell if you've been granted perms to edit PRs. It seems you can find out by opening the online text editor and if you don't have perms it'll tell you that you'd have to make changes in a fork. That's kind of dumb, but OK. It looks like you've given me those perms. Is it good etiquette to use them? Rather than just making changes you don't know about and merging them I'd rather be opening a PR to your PR, and then if you see my changes and agree with them we could merge both at the same time. I've never tried doing that though.

Reply:
I'm totally okay if you push to my fork, I didn't do anything to grant you the permission so I guess it's Github's default policy.

> A `Set` is used to do this check in constant time

This is not accurate, the `has()` method of `Set` works in logarithmic time (`O(logn)`), not constant time (`O(1)`).

Reply:
What's your source for `logn` time?

Reply:
Sorry, that's embarrassing, I have to take that one back. I do understand that typical (Hash)Set implementations have a time complexity of `O(1)`, but somehow I remember someone has made a benchmark and proved that it was actually `O(logn)`. But now I turn to look for that post I can't find it. Guess my memory betrayed me!

Reply:
I mean that's the thing with JS -- none of the perf is guaranteed. The spec doesn't say one way or the other. I don't actually have any evidence for `O(1)` either, but that's what I was taught about how hash maps should work. Anyway OK, let's update the docs for now and push this out so you can use it.

Reply:
Yeah specs say it's sublinear but I guess it's not a good idea to use that word in the docs.

Are you gonna make the changes or should I?

Reply:
I'll do it tomorrow

Reply:
ping